src/lib/homestore.cpp

```diff
@@ -231,6 +231,11 @@ void HomeStore::shutdown() {
 
     LOGINFO("Homestore shutdown is started");
 
+    if (has_repl_data_service()) {
+        s_cast< GenericReplService* >(m_repl_service.get())->stop();
+        m_repl_service.reset();
+    }
+
     if (has_index_service()) {
         m_index_service->stop();
```

Comment:
when indexservice stops, it will call CPManager to flush the CP, but RaftReplDev cp flush will crash if we reset the replicationservice in L236 before.

Reply:
Raised an issue https://github.com/eBay/HomeStore/issues/263 to track this issue.